### PR TITLE
[25490] Add publicity settings for query saving modal

### PR DESCRIPTION
**Pull request by oliverguenther**
When copying queries, the publicity for other members is copied, but not
the query `star` setting (whether it is shown in the project menu or not).

This is confusing to the user in two ways:

1. The user does not know whether a query is public or not when copying it

2. It is weird to copy one setting, but not the other

This PR suggests to extend the `save as` modal with the visibility settings. To this end, I extracted the visibility form into a separate directive to be re-used from the publish/sharing modal and the save as modal.

**Note:** When copying an existing query, both settings (star, public) are not copied over. This is because the user should think about which queries she wants to make public and available to the sidebar.

bildschirmfoto 2017-07-06 um 16 57 23

It also renames 'Share...' with 'Publish ...'  since sharing isn't really the right terminology.

https://community.openproject.com/projects/openproject/work_packages/25490

### CODE REVIEW

File: config/locales/js-en.yml [hunk 1]
```diff
@@ -539,6 +540,7 @@ en:
         save_as: "Save as ..."
         export: "Export ..."
         share: "Share ..."
```
Comment (ulferts) [inline config/locales/js-en.yml]: This translation key can probably be removed then.

**Review by ulferts — CHANGES_REQUESTED** [review]
### General
I for one like the change. I never liked having the query added automatically, and even worse, copying over the visibility setting. There where too many queries I have seen copied by others that where not intended to be shared.

But opinions can differ here, so maybe @lindenthal and @RobinWagner  can provide their opinion as well. 

### Bug:
image

When saving the query initially, I cannot select the "show page in menu" option

**Comment by oliverguenther** [comment]
This idea originated from a discussion with @wielinde , so I can safely assume he's on the pro side for this change.

**Comment by wielinde** [comment]
Love it!

**Comment by oliverguenther** [comment]
Fixed the bug, it was caused by the default query not having star/unstar permissions. I now assume that if you're able to save the query, you can star it as well (which the permission currently reflects).

**Review by ulferts — APPROVED** [review]

**Comment by RobinWagner** [comment]
I like the change. Makes a lot of sense.

**Comment by lindenthal** [comment]
@oliverguenther your change makes it much easier to understand. I like it.

**Pull request merged**